src/Containers/Impl/Docker/LocalDockerService.cs

```diff
@@ -77,86 +91,93 @@ public abstract class LocalDockerService : IDockerService {
         }
 
-        public Task<string> StartContainerAsync(IContainer container, CancellationToken ct) {
-            var command = "start";
-            var commandOptions = $"{container.Id}";
-            return ExecuteCommandAsync(Invariant($"{command} {commandOptions}"), -1, true, ct);
+        public Task<string> StartContainerAsync(string containerId, CancellationToken ct) {
+            var command = Invariant($"start {containerId}");
+            return ExecuteCommandAsync(command, command, -1, true, ct);
         }
 
-        public Task<string> StopContainerAsync(IContainer container, CancellationToken ct) {
-            var command = "stop";
-            var commandOptions = $"{container.Id}";
-            return ExecuteCommandAsync(Invariant($"{command} {commandOptions}"), -1, true, ct);
+        public Task<string> StopContainerAsync(string containerId, CancellationToken ct) {
+            var command = Invariant($"stop {containerId}");
+            return ExecuteCommandAsync(command, command, -1, true, ct);
         }
 
         protected abstract LocalDocker GetLocalDocker();
 
-        private async Task<string> ExecuteCommandAsync(string arguments, int timeoutms, bool failOnTimeout = true, CancellationToken ct = default(CancellationToken)) {
+        private async Task<string> ExecuteCommandAsync(string arguments, string outputPrefix, int timeoutms, bool failOnTimeout = true, CancellationToken ct = default(CancellationToken)) {
+            var printOutput = outputPrefix != null;
+            if (_output == null && printOutput) {
+                _output = await _outputService.GetAsync(ContainerOutputName, ct);
+            }
+
             await TaskUtilities.SwitchToBackgroundThread();
 
             var docker = GetLocalDocker();
-            ProcessStartInfo psi = new ProcessStartInfo {
+            var psi = new ProcessStartInfo {
+                CreateNoWindow = true,
                 FileName = docker.DockerCommandPath,
                 Arguments = arguments,
                 RedirectStandardError = true,
                 RedirectStandardInput = true,
                 RedirectStandardOutput = true,
-                UseShellExecute = false,
-                CreateNoWindow = true
+                UseShellExecute = false
             };
 
             var process = _ps.Start(psi);
+            var result = new StringBuilder();
             try {
+                while (!process.StandardOutput.EndOfStream) {
+                    var line = await process.StandardOutput.ReadLineAsync();
+                    if (printOutput) {
+                        _output.WriteLine(Invariant($"{outputPrefix}> {line}"));
+                    }
+                    result.AppendLine(line);
+                }
+
                 await process.WaitForExitAsync(timeoutms, ct);
             } catch(OperationCanceledException) when (!failOnTimeout && !ct.IsCancellationRequested){
```

Comment:
You may want to catch `IOException` just in case `process.StandardOutput` reading fails.

Reply:
Ok